user/client_test.go

```diff
@@ -0,0 +1,375 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+	"github.com/clerk/clerk-sdk-go/v2/clerktest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserClientCreate(t *testing.T) {
+	t.Parallel()
+	id := "user_123"
+	username := "username"
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			In:     json.RawMessage(fmt.Sprintf(`{"username":"%s"}`, username)),
+			Out:    json.RawMessage(fmt.Sprintf(`{"id":"%s","username":"%s"}`, id, username)),
+			Method: http.MethodPost,
+			Path:   "/v1/users",
+		},
+	}
+	client := NewClient(config)
+	user, err := client.Create(context.Background(), &CreateParams{
+		Username: clerk.String(username),
+	})
+	require.NoError(t, err)
+	require.Equal(t, id, user.ID)
+	require.Equal(t, username, *user.Username)
+}
+
+func TestUserClientList_Request(t *testing.T) {
+	t.Parallel()
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			Method: http.MethodGet,
+			Query: &url.Values{
+				"limit":         []string{"1"},
+				"offset":        []string{"2"},
+				"order_by":      []string{"-created_at"},
+				"email_address": []string{"foo@bar.com", "baz@bar.com"},
+			},
+		},
+	}
+	client := NewClient(config)
+	params := &ListParams{
+		EmailAddresses: []string{"foo@bar.com", "baz@bar.com"},
+		OrderBy:        clerk.String("-created_at"),
+	}
+	params.Limit = clerk.Int64(1)
+	params.Offset = clerk.Int64(2)
+	_, err := client.List(context.Background(), params)
+	require.NoError(t, err)
+}
+
+func TestUserClientList_Response(t *testing.T) {
+	t.Parallel()
+	usersJSON := `[{"object":"user","id":"user_123"}]`
+	countJSON := `{"object":"total_count","total_count":5}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.Contains(r.URL.Path, "count") {
+			_, err := w.Write([]byte(countJSON))
+			require.NoError(t, err)
+			return
+		}
+		_, err := w.Write([]byte(usersJSON))
+		require.NoError(t, err)
+	}))
+	defer ts.Close()
+
+	config := &ClientConfig{}
```

Comment:
🙃 Struct fields can be assigned here directly.

Reply:
Yes, but it saves me a bit of typing because the fields belong to the embedded struct :smile: :smile:

Reply:
Oh I see. Not sure if we want to optimise on initialization, just a thought for the nice-to-have TODO list 😄

Reply:
Yes, I thought about it as well. I cannot imagine these config options to go beyond the BackendConfig any time soon, so a helper might make sense. It will last long enough :smile: